### PR TITLE
doc: releases: v3.5.0: Add Bluetooth release notes

### PR DESCRIPTION
**Pull request by cvinayak**
Add Bluetooth release notes for Zephyr v3.5.0.

### CODE REVIEW

File: doc/releases/release-notes-3.5.rst
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:100]: Any chance you could have a couple bullet point with high-level summary (thinking 3-5 entries) of what are the most notable changes across all bluetooth? It's kind of difficult to figure it out by looking at the pretty large/flat list of bullet items.

**Comment by jhedberg** [comment]
@cvinayak is there anything that would need to go to the migration guide? (i.e. stuff that will require out-of-tree apps to update their code)

Btw, there seems to be a CI failure

File: doc/releases/release-notes-3.5.rst
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:196]: do you really want/need to go to this level of detail? (this probably applies to several other entries but this is one that caught my eye)

**Comment by cvinayak** [comment]
> @cvinayak is there anything that would need to go to the migration guide? (i.e. stuff that will require out-of-tree apps to update their code)

@jori-nordic  @Thalley  and @PavelVPV is there anything in host, audio and mesh needing update to migration doc?

> 
> Btw, there seems to be a CI failure

I do not understand the error, is the documentation trying to add URL/Link to symbols in the text?

**Review by kartben — CHANGES_REQUESTED** [review]
Some early and general feedback, in addition to the couple review comments, before I have a closer look:
- look at sorting list alphabetically, which will also have the benefit of making the "added" entries show up first
- consistently use the past tense (ex. "Fix"-->"Fixed")
- the granularity for some of the fixes seems to be very very specific, so maybe you can look at grouping some of the entries and focus on the impact of the end users. 
- several entries seem to be better suited for the migration guide, ex. it looks like some Kconfig options may have been removed: what's expected from users to adapt their code?

Hope this helps. Thank you!

File: doc/releases/release-notes-3.5.rst
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:109]: please try to use double backquotes for anything that should be formatted as code

```suggestion
  * Renamed ``bt_codec`` to ``bt_audio_codec_{cap, conf, data}``
```

**Comment by Thalley** [comment]
> is there anything in host, audio and mesh needing update to migration doc?

Quite a bit actually, since we had a few refactors in the LE Audio API :( 

Are we doing migration guides for experimental APIs?

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [reply doc/releases/release-notes-3.5.rst:109]: Is that the right way of doing it in RST?

In most RST files I've seen (and written), we've used 
:code:\`bt_codec\` instead

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [inline doc/releases/release-notes-3.5.rst:110]: This is not Audio related

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [inline doc/releases/release-notes-3.5.rst:120]: These are ISO related, not Audio

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [reply doc/releases/release-notes-3.5.rst:196]: Do we have any guidelines of what goes into the release notes, and what does?

I assume anything that changes behavior, API or bug fixes

**Review by Thalley — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [reply doc/releases/release-notes-3.5.rst:110]: Github isn't great, but it is specifially this line

```
Fixed OTS bt_ots_init parameter struct naming
```

That isn't audio related

File: doc/releases/release-notes-3.5.rst
Comment (kartben) [reply doc/releases/release-notes-3.5.rst:109]: Either way should be fine bust a vast majority of our docs uses double backticks (and that's what's also recommended in the [guidelines](https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#content-highlighting)).

Comment (cvinayak) [reply doc/releases/release-notes-3.5.rst]: fixed

File: doc/releases/release-notes-3.5.rst
Comment (cvinayak) [reply doc/releases/release-notes-3.5.rst]: Moved

File: doc/releases/release-notes-3.5.rst
Comment (cvinayak) [reply doc/releases/release-notes-3.5.rst]: Moved

File: doc/releases/release-notes-3.5.rst
Comment (cvinayak) [reply doc/releases/release-notes-3.5.rst:100]: I have added couple lines now under Controller.

@Thalley @jori-nordic @PavelVPV may be you can give some lines and I will include them...

Comment (Thalley) [reply doc/releases/release-notes-3.5.rst:100]: For LE audio you can add something like

Improved memory usage of codec configurations codec capabilities. Fixed several bugs in BAP and the BAP-related services (ASCS, PACS, BASS), as well as missing features such as proper notification handling.

**Review by Thalley — CHANGES_REQUESTED** [review]

**Comment by jhedberg** [comment]
> Are we doing migration guides for experimental APIs?

I actually don't know. My initial thought is that the migration doc might get quite noisy. Does @kartben or @carlescufi have some guidance on this?

File: doc/releases/release-notes-3.5.rst
Comment (PavelVPV) [inline doc/releases/release-notes-3.5.rst:191]: Thanks for collecting these notes! I tried to group and rephrase them a bit:
```
  * Added TF-M support.
  * Added support to use both tinycrypt and PSA based crypto
  * Added full virtual addresses support with the collision resolution. The :kconfig:option:`CONFIG_BT_MESH_LABEL_NO_RECOVER` Kconfig option is introduced to restore the addresses for the subscription list and model publication.
  * Added statistic module.
  * Fixed an issue where a node acting as a LPN was triggering Friend Poll messages when sending a segmented message over the loopback interface.
  * Fixed an issue where provisioning completes successfully on a node when the identical Public Key is used by a  provisioner.
  * Fixed an issue where the :c:func:`settings_load` function called from a cooperative thread other than the system workqueue caused the GATT Mesh Proxy Service registration to fail.
  * Fixed an issue where a node could enter IV Update in Progress state if an old SNB with the current IV Index and IV Update flag set to 1 was resent.
  * Fixed for Coverity reported issues.
  
  * Mesh Protocol v1.1 changes:
    * Added storing Private GATT Proxy state persistently.
    * Added support for Firmware Distribution Upload OOB Start message in the Firwmware Distribution Server model. The message support can be enabled with the :kconfig:option:`CONFIG_BT_MESH_DFD_SRV_OOB_UPLOAD` Kconfig option.
    * Added extended provisioning protocol timeout when OOB methods are used in the provisioning.
    * Added support for Composition Data Pages 2, 129 and 130.
    * Added documentation for Composition Data Pages 0, 1, 2, 128, 129 and 130.
    * Added documentation for the Segmentation and Reassembly in the Transport layer.
    * Added documentation for the SAR Configuration models
    * Fixed an issues where the Opcode Aggregator Server model did not compile without the Opcode Aggregator Client model.
    * Fixed an issue where the identity address was used in Private GATT Proxy advertisements instead of Non-Resolvable Private Addresses.
    * Fixed the Proxy Privacy parameter support.
    * Fixed an issue where the Composition Data Page 128 was not present on a node that has instantiated the Remote Provisioning Server model.
    * Fixed an issue where the Large Composition Data Server model did not support Composition Data Pages other then 0.
    * Fixed an issue where the Remote Provisioning Client model instanted on a node together with the Remote Provisioning Server model could not reprovision itself.
    * Fixed an issue where the acknowledgement timer in the Segmentation and Reassembly was not restarted when the incoming Segment Acknowledgement message did not contain at least one segment newly marked as acknowledged.
    * Fixed an issue where the On-Demand Private Proxy Server and Client models had interdependency that did not allow to compile them separately.
```

**Review by PavelVPV — DISMISSED** [review]

File: doc/releases/release-notes-3.5.rst
Comment (Thalley) [inline doc/releases/release-notes-3.5.rst:103]:
```suggestion
  Improved memory usage of codec configurations and codec capabilities. Fixed several bugs in BAP and
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor request

**Comment by jhedberg** [comment]
@cvinayak @carlescufi let's try to get this to a mergeable state today. What's the status with the migration guide changes?

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
This looks ok to me for an initial version. I'd propose to merge it and then work on smaller improvements (if necessary) and the migration guide in a separate PR. The migration guide is anyway getting a reshuffle as part of #64074 so might be best to wait until that's merged.

**Review by kartben — APPROVED** [review]
> This looks ok to me for an initial version. I'd propose to merge it and then work on smaller improvements (if necessary) and the migration guide in a separate PR. The migration guide is anyway getting a reshuffle as part of https://github.com/zephyrproject-rtos/zephyr/pull/64074 so might be best to wait until that's merged.

+1

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**
